public/app/features/trails/MetricSelectScene.tsx

```diff
@@ -131,8 +204,36 @@ export class MetricSelectScene extends SceneObjectBase<MetricSelectSceneState> {
     this.state.body.setState({ children, autoRows: rowTemplate });
   }
 
+  private shouldRebuildLayout = () => {
+    return !Object.values(this.state.previewMetricPanels).some((mp) => mp.isPanel && !mp.loaded);
+  };
+
+  public hideMetric = (metric: string) => {
+    const metricPanel = this.state.previewMetricPanels[metric];
+    if (metricPanel) {
+      metricPanel.isHidden = true;
+      metricPanel.loaded = true;
+      this.setState({ previewMetricPanels: { ...this.state.previewMetricPanels, [metric]: metricPanel } });
+      if (this.shouldRebuildLayout()) {
+        this.buildLayout();
+      }
+    }
+  };
+
+  public metricHasLoaded = (metric: string) => {
+    const metricPanel = this.state.previewMetricPanels[metric];
+    if (metricPanel) {
+      metricPanel.loaded = true;
+      this.setState({ previewMetricPanels: { ...this.state.previewMetricPanels, [metric]: metricPanel } });
+      if (this.shouldRebuildLayout()) {
+        this.buildLayout();
+      }
+    }
+  };
```

Comment:
not sure we need all this as the layout system has a built-in hide feature (state).